app/src/main/java/ch/epfl/sdp/blindwar/ui/VoskActivity.kt

```diff
@@ -54,6 +59,36 @@ class VoskActivity : Activity(), RecognitionListener {
         } else {
             initModel()
         }
+        //recognizer with google
+        voiceRecognizer.init(this, resultView as TextView)
+
+        findViewById<View>(R.id.recognize_mic_google).setOnTouchListener { v, event ->
+            when (event?.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    voiceRecognizer.start()
+                }
+                MotionEvent.ACTION_UP -> {
+                    v?.performClick()
+                    voiceRecognizer.stop()
+                }
+            }
+            v?.onTouchEvent(event) ?: true
+        }
+    }
+
+    override fun onActivityResult(
+        requestCode: Int, resultCode: Int,
+        data: Intent?
+    ) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == 1) {
+            if (resultCode == RESULT_OK && data != null) {
+                val result = data.getStringArrayListExtra(
+                    RecognizerIntent.EXTRA_RESULTS
+                )
+                resultView?.text = result!![0]

```

Comment:
I'm not sure but I feel like sometimes using !! is unsafe(sometimes you should probably treat the error